### ISSUE TITLE
Allow "led-row-address-type" for FAST RPi Models.

### ISSUE THREAD

**Issue by mortification77** [FalconChristmas/fpp#1675]
FPP v7.3.1 running on RPi 4 (4MB RAM)

FPP is missing the "led-row-address-type" Option that was added to https://github.com/hzeller/rpi-rgb-led-matrix shortly after FPP was built.

Add the option to FPP.

(Edited my ORIGINAL COMMENT with accusations of disabling the feature -- sorry lol).

**Comment by cpinkham** [comment]
What line of FPP code are you referring to?  As far as I can tell, FPP doesn't touch the row_address_type option, but that doesn't mean we disabled something.  If you would like to submit a patch to support the backend and UI functionality to set this to the various options, we'd be more than willing to look at a pull request.

**Comment by mortification77** [comment]
I have not dug into the code yet, but this is what I’m talking about:

**Comment by mortification77** [comment]
@cpinkham I guess I should have analyzed a bit more.

I ASSumed that it was not available on RPi because it was explicitly disabled!

However looking at the plugins, it is likely due to the fact that FPP on the RPi and BBB use separate plugins for LED Panels and it was simply NEEVER ADDED when the option became available in the Lib for Pi.

I would be delighted to facilitate this!!!

**Comment by mortification77** [comment]
I have a working patch to add the led-row-address-type to FPP for RPi.

@cpinkham My repo knowledge is very limited, so may need some Pull Request guidance!

**Comment by darylc** [comment]
@mortification77 how did you go with your pull request?

**Comment by mortification77** [comment]
> @mortification77 how did you go with your pull request?

I am totally new to doing this.

I have my changes down to a patch file that I have successfully applied to my local version.  If I clone the main branch, I can apply it and run it with no problems.

If I FORK the branch, apply my changes, for some reason it complains about missing files and doesn't work or run.

Some help would be awesome!

**Comment by mortification77** [comment]
> @mortification77 how did you go with your pull request?

@cpinkham and @darylc 

Do I need to assign the Pull Request to someone or did I do it correctly?

**Comment by OnlineDynamic** [comment]
@mortification77 can this be closed now? Looks like PR was merged successfully... is there anything else outstanding?

**Comment by mortification77** [comment]
All set.

Thanks for getting my mod in and dealing with my fumbling through my first PR.

This option that we added was added to the matrix library shortly AFTER FPP's original repo date, so that is why FPP didn't have it.

I used the matrix library in my own projects, so I know it quite well.

I believe it is the only option that FPP was missing that relates to internal panel mapping.

I will EVENTUALLY be testing out direct GPIO mapping with new style panels that I have had difficulty driving.  HOWEVER, I was successful with an ESP32 with direct GPIO to HUB75 wiring, so I'm hopeful that I will be successful with the GPIO wiring per the matrix library's guide.

If more options are needed, we may revisit this again.

Thanks again!
